SerenityStation - New Planetary Station Map

Thlumyn opened the pull request:
## About The Pull Request

Big thanks to GoldenAlpharex for all his help on the code & biome generation.

A nova-specific map featuring a station located on a forest planet. The forest planet has caves filled with mushrooms, and underground cold liquid plasma springs bubbling to the surface.

This includes 3 z-levels: the underground mushroom level; the surface forest level; and a space level that has a small orbital station and linkage to other space z levels.

This station features a custom shuttle to go between the orbital and planetary shuttle, though the public mining shuttle can also go between them (with 3 docking options now: planetary dock, orbital dock and lavaland).

There are custom mushroom tree, mushroom floor, mushroom flora, stillcap and megadeer sprites for this map. The megadeer is more or less a reskinned wolf and drops a reskinned wolf crusher trophy. The stillcap is a basic mob that hides as a mushroom when a target is not nearby.

There are two main cave generation types: the surface, which functions as normal cave generation such as icebox, and the lower mushroom level which uses biome generation to mix in three different biomes (red, blue and green)

The name comes from the ship Serenity from Firefly, the show that got me into science fiction.
## How This Contributes To The Nova Sector Roleplay Experience

New maps are kind of fun. This one has planetary areas for people to explore which provide new mobs, new plants and new turfs. It also has an orbital station so that people can still spacetide and space ghost roles can still play on this map. People have been asking for some non-icebox planetary maps for a while.
## Proof of Testing
<details>
<summary>Screenshots/Videos</summary>
  
![image](https://github.com/NovaSector/NovaSector/assets/102194057/0cdf432a-c341-4f40-bb01-f2e653b5d7cf)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/7dbb6a11-c390-4545-a766-58c744ff311a)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/e86003ad-4d4d-48da-9139-59143bc3989b)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/892cd0ca-d7db-4924-b4a7-859c8517b76b)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/e1ea9ee7-d65b-4229-afcf-a7c29673a3de)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/efac5f4e-f709-4554-983c-9b40bc5cf4af)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/4cf76798-68ba-4667-832d-f38e741423bf)

</details>

## Changelog
:cl: Thlumyn for the map and sprites, GoldenAlpharex for most of the code
add: Added SerenityStation map.
add: Added lightshroom, mushroom bush, mushroom tree, and colored ash flora mushroom plants.
add: Added Stillcap and Mega Deer mobs.
add: Add forest cave generation and mushroom cave biome generation.
config: Add SerenityStation to the map config.
/:cl:

MajManatee commented:
Mushroom

RunKittenzRComing commented:
Mushroom

Zergspower commented:
Here's a first mostly disposals pass, dont have time to do department checks atm

<Details>
<summary>Disposals</summary>
Floor 2 security station 98|131 - Missing disposals
![image](https://github.com/NovaSector/NovaSector/assets/22140677/c4f3088a-ec4e-44cb-ad48-fc54586e738b)

You'll need to redesign cargo's trunk system here, rebuilding this would be extremely difficult for a number. I'd either turn them so they share the belt space or extend a pipe 1 more out into the lobby
![image](https://github.com/NovaSector/NovaSector/assets/22140677/5b5aac9e-f0bc-4d7a-ac8d-c7e47b4ed171)

There are also a number of disposals crossing through walls, try to avoid doing this unless absolute necessary for the reason above, pipes can break and it needs to make sense to quickly repair or people wont bother. I suggest looking at DeltaStation for ideas on how to handle multiples 
![image](https://github.com/NovaSector/NovaSector/assets/22140677/70fd9854-9096-4405-8b97-97f1d6c17566)
![image](https://github.com/NovaSector/NovaSector/assets/22140677/b904c8b2-065e-4657-871b-ad6130a91672)
![image](https://github.com/NovaSector/NovaSector/assets/22140677/5e465f7f-5dd1-4008-b8da-96cb340dff55)
![image](https://github.com/NovaSector/NovaSector/assets/22140677/3ede9a54-e785-417c-83ef-9522e2f83945)
![image](https://github.com/NovaSector/NovaSector/assets/22140677/13ecdbf1-d11a-4a53-b642-a77db3662062)

</details>

<Details>
<summary>Cable - Power</summary>

Kitchen has cables running across but don't connect to the main grid, these can probably get deleted
![image](https://github.com/NovaSector/NovaSector/assets/22140677/d0e726fd-860e-420e-8d69-d7595cea1bef)

Level 3 is goin wild, lower stray cables and - what's goin on in the utilities room
![image](https://github.com/NovaSector/NovaSector/assets/22140677/33b8775c-0a1e-4f30-9787-10fba4103a78)
</details>

Thlumyn commented:
As far as the utilities room, that is to avoid the cables from the grid connecting to the solar input cables into the SMES

![image](https://github.com/NovaSector/NovaSector/assets/102194057/4bc1649d-5670-4f2c-8f90-de3617a9375e)

Thlumyn commented:
Okay, I've tried to fix some of them 

<details>
<summary>Disposals moved to be not in wall</summary>

![image](https://github.com/NovaSector/NovaSector/assets/102194057/30c1bba6-f199-47ac-80c9-48c454f1675c)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/c8961882-b0b3-4c9e-8c70-15c024146c35)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/608afe3e-f825-46ef-aebd-09adcdaf47e4)

![image](https://github.com/NovaSector/NovaSector/assets/102194057/6620b392-8517-415c-b383-898b4aaa0123)
</details>

This one is for the corpse disposal; on IceBox it goes through the wall. Honestly, I don't often see this one used, and I can't find a good way for it to not go through a wall so if it can't go like this I'll probably rework the morgue to remove it. 

<details>
<summary>Corpse disposal</summary>

![image](https://github.com/NovaSector/NovaSector/assets/102194057/fd391aaa-6ec0-4eae-91ee-fbd4c1c6cc06)

</details>

The up/down trunks going into a wall is taken out of tram; if that needs to be changed I can, it might be a bit though as it would mean reworking the disposals room. 

As far as the departmental chutes going through walls; on Meta and Void it also goes through walls like. On Blueshift it goes through windows there it looks like. Would windows be preferable?

Zergspower commented:
Given it's the morgue we can handwaive that one - i'm thinking strictly from the 'fixing' perspective as pipes can break before walls/tiles can. Funny while looking at Delta to check something i found 3 stray pipes on their side fun.

I'll give it another pass here when im able to sit down and not feel like a knife is stabbing me

TheSS13Melon commented:
mushrooms scare me, can we make the map less scary

Thlumyn commented:
@vinylspiders  I've made those changes to the turf/flora/etc code
I'll have to ask @GoldenAlpharex  to take a look at the stillcap/ai code,

Jolly-66 commented:
Hi. Pardon the draft. Doing this preemptively so we (the maptainers) have sufficient time to dissect and look at your map. 

Thank you for your understanding!

GoldenAlpharex commented on modular_nova/modules/serenitystation/code/mobs/stillcap/stillcap.dm:
```diff
@@ -0,0 +1,78 @@
+	AddComponent(/datum/component/basic_mob_attack_telegraph, telegraph_duration = 0.6 SECONDS)
+
+
+// /mob/living/basic/mining/stillcap/toggle_hiding(new_hiding)
+// 	. = ..()
+// 	pass_flags_self = hiding ? PASSSTRUCTURE : PASSMOB
+
+
+// /mob/living/basic/mining/hiding/update_icon_state()
+// 	. = ..()
+// 	icon_state = "[base_icon_state][hiding ? "_hidden" : ""]"
+
+
```
Nope.

GoldenAlpharex commented on code/datums/ai/basic_mobs/basic_ai_behaviors/targeting.dm:
```diff
@@ -71,7 +71,7 @@ GLOBAL_LIST_INIT(target_interested_atoms, typecacheof(list(/mob, /obj/machinery/
 	var/aggro_range = controller.blackboard[aggro_range_key] || vision_range
 	// takes the larger between our range() input and our implicit hearers() input (world.view)
-	aggro_range = max(aggro_range, ROUND_UP(max(getviewsize(world.view)) / 2))
+	// aggro_range = max(aggro_range, ROUND_UP(max(getviewsize(world.view)) / 2)) // NOVA EDIT REMOVAL - This messes up the range of the stillcaps, will be possibly brought upstream as a result.
```
I don't think that it will, but we can probably test it out and see if that causes any issues.

There's not really any other solution to this, unless I was to duplicate a lot of code to not do this. The only changes this would make is that mobs would only really wake up when you get in their aggro range, instead of as soon as you get within screen distance of them.

GoldenAlpharex commented on modular_nova/modules/serenitystation/code/mobs/ai/ai_behaviors.dm:
```diff
@@ -0,0 +1,57 @@
+		controller.set_blackboard_key(BB_HIDING_AGGRO_RANGE_NOT_HIDING, controller.blackboard[BB_AGGRO_RANGE])
+
+
+/datum/ai_behavior/toggle_hiding/perform(seconds_per_tick, datum/ai_controller/controller, now_hiding)
```
It does not, but this sounds like *such* a niche case that I don't think it's really worth accounting for. They're going to end up getting up anyway, eventually.

Krypandenej commented:
God DAMN this map is GORGEOUS! All we need is some good ambiance for the outside and mushroom caverns...

https://youtu.be/5UZE-Ln2-Eg?si=FUC7rFjorOOF1OFB

GoldenAlpharex commented:
> God DAMN this map is GORGEOUS! All we need is some good ambiance for the outside and mushroom caverns...
> 
> https://youtu.be/5UZE-Ln2-Eg?si=FUC7rFjorOOF1OFB

There's already ambience coming from the mushroom caverns, I just wanted to make the PR separately to give the ambience's author the recognition they deserve.

Zergspower commented:
I'm a bit late for the engineering review and sorry about that, Birdshot PR's took my will to live and threw it out the window. 

I solicited feedback from a few engineers who have managed to play a shift and asked their opinions, and the common one was how small and cramped engineering is. And looking at it now closer i can agree as it's just a small hallway and that's basically it outside of the lobby and the SM room.
![image](https://github.com/NovaSector/NovaSector/assets/22140677/1c758869-2322-434a-9d45-9579052568c9)

General movement seems quite restricted here, but i think the only solution without redoing that entire section is to move the CE's office and turn that into their more general space - but i'd have to play with the mapper to really see how it'd work but i do agree that something's needing altered here. Or instead maybe give them an underground area to alleviate it as there's open space below.

Atmos from what i'm told is fine and looking at it i cant see any glaring issues popping out while setting atmos up IC, so i think we're good here.

<Summary>
<details>walls</details>
NTC/BS
Everything but this one small part is reinforced, not sure if that was an oversight or not

![image](https://github.com/NovaSector/NovaSector/assets/22140677/e0b8908e-82bc-4d3d-b768-3bcc27138c79)

Similar thing here with the detectives office

![image](https://github.com/NovaSector/NovaSector/assets/22140677/13bc520c-9989-4155-93a2-d217c760d4f4)

Xenobio's airlock has an extra R-wall jutting out on the right here

![image](https://github.com/NovaSector/NovaSector/assets/22140677/7a88bc60-fb4b-4300-a03d-fe5eca7a2ae8)

Same in Ord Maints

![image](https://github.com/NovaSector/NovaSector/assets/22140677/827c0a3c-b0d9-4b0e-9184-6c7145ff161d)

Command stairwell

![image](https://github.com/NovaSector/NovaSector/assets/22140677/127fb3d3-0922-4f28-aad2-8fcac02a07e4)

Right side of Bridge Maints wall

![image](https://github.com/NovaSector/NovaSector/assets/22140677/04b1c206-8443-4bce-bf47-c8de2b3ce517)

2 extra walls in the bottom right here in the HFR room

![image](https://github.com/NovaSector/NovaSector/assets/22140677/dff9c7c6-0c61-48fd-825e-6d8a1ccfc0fb)
</details>

Thlumyn commented:
@Zergspower  Okay, I moved the CE's office to the right, taking a bit of atmos's room, but widening the area for engineering. 
I addressed the rwall issues

GoldenAlpharex reviewed (approved):
Should be good to go now, it's been tested enough. This will finally allow us to add more stuff to it without stealing the thunder from this awesome PR (I'm totally not biased here).
